AWSAppSyncClient/AWSAppSyncClient.swift

```diff
@@ -918,13 +930,88 @@ public class AWSAppSyncClient {
+
+        // Safe to force-unwrap here; we only care about the EmptyQuery's `operationString`
+        let resolvedDeltaQuery: DeltaQuery = deltaQuery ?? EmptyQuery() as! DeltaQuery
+        let resolvedDeltaCallback: DeltaQueryResultHandler<DeltaQuery> = deltaQueryResultHandler ?? { (_, _, _) in }
+
+        return AppSyncSubscriptionWithSync<EmptySubscription, BaseQuery, DeltaQuery>(appSyncClient: self,
+                                                                                     baseQuery: baseQuery,
+                                                                                     deltaQuery: resolvedDeltaQuery,
+                                                                                     subscription: subscription,
+                                                                                     baseQueryHandler: baseQueryResultHandler,
+                                                                                     deltaQueryHandler: resolvedDeltaCallback,
+                                                                                     subscriptionResultHandler: subscriptionResultHandler,
+                                                                                     subscriptionMetadataCache: self.subscriptionMetadataCache,
+                                                                                     subscriptionWithSyncConfiguration: resolvedSubscriptionWithSyncConfiguration,
+                                                                                     handlerQueue: callbackQueue)
+    }
+
+    /// Performs a sync operation where a base query is periodically called to fetch primary data from the server based on the syncConfiguration.
+    ///
+    /// - Parameters:
+    ///   - baseQuery: The base query to fetch which contains the primary data.
+    ///   - baseQueryResultHandler: Closure that is called when base query results are available or when an error occurs. Every time a sync operation is called, a fetch for the baseQuery from the cache will be done first before initiating any other operations.
+    ///   - subscription: The subscription which will provide real time updates.
+    ///   - subscriptionResultHandler: Closure that is called when a real time update is available or when an error occurs.
+    ///   - deltaQuery: The delta query which fetches data starting from the `lastSync` time.
+    ///   - deltaQueryResultHandler: Closure that is called when delta query executes.
+    ///   - callbackQueue: An optional queue on which sync callbacks will be invoked. Defaults to the main queue.
+    ///   - subscriptionWithSyncConfiguration: The sync configuration where the baseQuery sync interval can be specified. (Defaults to 24 hours.)
+    /// - Returns: An object that can be used to cancel the sync operation.
+    public func sync<BaseQuery: GraphQLQuery, Subscription: GraphQLSubscription, DeltaQuery: GraphQLQuery>(baseQuery: BaseQuery,
```

Comment:
I think that parameter `subscriptionWithSyncConfiguration` can just be named `configuration` or `config`. It's already domain specific to the function `sync`.